lib/compat/wordpress-6.5/navigation-block-hooks.php

```diff
@@ -0,0 +1,119 @@
+<?php
+add_filter( 'rest_prepare_wp_navigation', 'gutenberg_hook_blocks_into_core_navigation_rest_response', 10, 3 );
+add_filter( 'block_core_navigation_render_inner_blocks', 'gutenberg_hook_blocks_into_core_navigation', 10, 3 );
+add_action( 'rest_insert_wp_navigation', 'gutenberg_update_ignore_hooked_blocks_meta_core_navigation', 10, 3 );
+
+/**
+ * Updates the post meta with the list of ignored hooked blocks when the navigation is created or updated via the REST API.
+ *
+ * @param WP_Post $post Post object.
+ */
+function gutenberg_update_ignore_hooked_blocks_meta_core_navigation( $post ) {
+	if ( ! isset( $post->ID ) ) {
+		return;
+	}
+
+	$anchor_block_type    = 'core/navigation';
+	$first_child_position = 'first_child';
+	$last_child_position  = 'last_child';
+
+	$hooked_blocks                 = get_hooked_blocks();
+	$hooked_block_types_last_child = isset( $hooked_blocks[ $anchor_block_type ][ $last_child_position ] )
+	? $hooked_blocks[ $anchor_block_type ][ $last_child_position ]
+	: array();
+
+	$hooked_block_types_first_child = isset( $hooked_blocks[ $anchor_block_type ][ $first_child_position ] )
+	? $hooked_blocks[ $anchor_block_type ][ $first_child_position ]
+	: array();
```

Comment:
Nitpick: Alignment

```suggestion
	$hooked_block_types_last_child = isset( $hooked_blocks[ $anchor_block_type ][ $last_child_position ] )
		? $hooked_blocks[ $anchor_block_type ][ $last_child_position ]
		: array();

	$hooked_block_types_first_child = isset( $hooked_blocks[ $anchor_block_type ][ $first_child_position ] )
		? $hooked_blocks[ $anchor_block_type ][ $first_child_position ]
		: array();
```

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/57754/commits/8e5a19fc1f135ca381bffd59c512fc94f508dcd1